outdated/modules/httpd.reb

```diff
@@ -225,199 +256,221 @@ sys/make-scheme [
+            200 "OK"
+            201 "Created"
+            204 "No Content"
+            
+            301 "Moved Permanently"
+            302 "Moved temporarily"
+            303 "See Other"
+            307 "Temporary Redirect"
+            
+            400 "Bad Request"
+            401 "No Authorization"
+            403 "Forbidden"
+            404 "Not Found"
+            411 "Length Required"
+
+            500 "Internal Server Error"
+            503 "Service Unavailable"
+        ])
+
+        build-header (function [response [object!]] [
             append make binary! 1024 spaced collect [
-                case/all [
-                    not find status-codes response/status [
-                        response/status: 500
-                    ]
-                    any [
-                        not find [binary! string!] to word! type-of response/content
-                        empty? response/content
-                    ][
-                        response/content: " "
-                    ]
+                if not find status-codes response/status [
+                    response/status: 500
+                ]
+                any [
+                    not match [binary! text!] response/content
+                    empty? response/content
+                ] then [
+                    response/content: " "
                 ]
 
-                keep ["HTTP/1.1" response/status select status-codes response/status]
+                keep ["HTTP/1.1" response/status
+                    <- select status-codes response/status]
                 keep [cr lf "Content-Type:" response/type]
                 keep [cr lf "Content-Length:" length-of response/content]
-                case/all [
-                    response/compress? [
-                        keep [cr lf "Content-Encoding:" "gzip"]
-                    ]
-                    response/location [
-                        keep [cr lf "Location:" response/location]
-                    ]
-                    response/close? [
-                        keep [cr lf "Connection:" "close"]
-                    ]
+                if response/compress? [
+                    keep [cr lf "Content-Encoding:" "gzip"]
+                ]
+                if response/location [
+                    keep [cr lf "Location:" response/location]
+                ]
+                if response/close? [
+                    keep [cr lf "Connection:" "close"]
                 ]
                 keep [cr lf cr lf]
             ]
-        ]
-
-        function [client [port!]][
-            client/locals/response: response: make response-prototype []
-            client/locals/parent/locals/handler client/locals/request response
+        ])
+    ][
+        client/locals/response: response: make response-prototype []
+        client/locals/parent/locals/handler client/locals/request response
 
-            if response/compress? [
-                response/content: compress/gzip response/content
```

Comment:
Non-Stylistic Change: COMPRESS is gone, for now.  Bugs and ambiguities related to it and its refinements have been worked out across the board.